### ISSUE TITLE
Error: invalid name "myhubusername/voting-app"

### ISSUE THREAD

**Issue by tullo** [docker/app#445]
**Description**

Deployment of examples/voting-app fails with: `Error: invalid name "myhubusername/voting-app". Names must be [a-zA-Z0-9-_]+`

**Steps to reproduce the issue:**
1. `cd examples/voting-app/`
2. `make deploy/development`

**Describe the results you received:**

```
make deploy/development
docker-app render --parameters-files voting-app.dockerapp/parameters/development.yml > build/development/docker-compose.yml
docker stack rm voting-app
Nothing found in stack: voting-app
docker-app deploy --parameters-files voting-app.dockerapp/parameters/development.yml
Error: invalid name "myhubusername/voting-app". Names must be [a-zA-Z0-9-_]+
Makefile:53: recipe for target 'deploy/development' failed
make: *** [deploy/development] Error 1
```

**Describe the results you expected:**

A successful deployment

**Output of `docker version`:**

```
Client:
 Version:           18.09.1
 API version:       1.39
 Go version:        go1.10.6
 Git commit:        4c52b90
 Built:             Wed Jan  9 19:35:31 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.1
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.6
  Git commit:       4c52b90
  Built:            Wed Jan  9 19:02:44 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Output of `docker-app version`:**

```
Version:      v0.7.3
Git commit:   332efaa
Built:        Fri Nov 30 15:17:53 2018
OS/Arch:      linux/amd64
Experimental: off
Renderers:    none
```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 81
Server Version: 18.09.1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 256
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: active
 NodeID: 9k3p09fn8w3xc7axkcmkhvfaa
 Is Manager: true
 ClusterID: 0h5e918ercyf52rr6seldwigb
 Managers: 1
 Nodes: 1
 Default Address Pool: 10.0.0.0/8  
 SubnetSize: 24
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 10
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
  Force Rotate: 0
 Autolock Managers: false
 Root Rotation In Progress: false
 Node Address: 192.168.0.126
 Manager Addresses:
  192.168.0.126:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 9754871865f7fe2f4e74d43e2fc7ccd237edcbce
runc version: 96ec2177ae841256168fcf76954f7177af9446eb
init version: fec3683
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.15.0-43-generic
Operating System: Ubuntu 18.04.1 LTS
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.55GiB
Name: anda-G750JS
ID: RVWR:ZFB7:TORC:AZFZ:JXQI:EQLX:H6K5:J6V5:FOV2:4WD3:RNFB:KTA4
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: tullo
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
Product License: Community Engine
```

**Comment by tullo** [comment]
Same deployment works as expected using **docker stack deploy**

`docker-app render --parameters-files voting-app.dockerapp/parameters/production.yml | docker stack deploy --compose-file - voting`

**Comment by ijc** [comment]
Hi @tullo, thanks for your report.

It looks like you are running with the CNAB preview which is not in mainline yet so might have issues.

I cannot reproduce with current master:
```console
$ git clean -fdqx
$ git rev-parse HEAD
3603491fdcc72ebf31195a5ba092e50628dbff38
$ make bin/docker-app
CGO_ENABLED=0 go build -tags="" -ldflags="-s -w -X github.com/docker/app/internal.GitCommit=3603491f -X github.com/docker/app/internal.Version=v0.6.0-44-g3603491f -X github.com/docker/app/internal.Experimental=off -X github.com/docker/app/internal.BuildTime=2019-01-23T15:15:33Z" -o bin/docker-app ./cmd/docker-app
$ cd examples/voting-app/
$ PATH=../../bin:$PATH make deploy/development
docker-app render --parameters-files voting-app.dockerapp/parameters/development.yml > build/development/docker-compose.yml
docker stack rm voting-app
Nothing found in stack: voting-app
docker-app deploy --parameters-files voting-app.dockerapp/parameters/development.yml
Creating network voting-app_backend
Creating network voting-app_frontend
Creating network voting-app_default
Creating service voting-app_result
Creating service voting-app_worker
Creating service voting-app_visualizer
Creating service voting-app_redis
Creating service voting-app_db
Creating service voting-app_vote
$ docker service ls
ID                  NAME                    MODE                REPLICAS            IMAGE                                          PORTS
j6waflrqo1xv        voting-app_db           replicated          1/1                 postgres:9.4                                   
d25z6zwxv6cg        voting-app_redis        replicated          1/1                 redis:alpine                                   *:30001->6379/tcp
8v5fylqm6iy9        voting-app_result       replicated          0/1                 result:latest                                  *:8181->80/tcp
lnpvzzgwick5        voting-app_visualizer   replicated          1/1                 dockersamples/visualizer:latest                *:8282->8080/tcp
rc3cnzl548ka        voting-app_vote         replicated          0/1                 vote:latest                                    *:8080->80/tcp
kzqd2otif4u1        voting-app_worker       replicated          1/1                 dockersamples/examplevotingapp_worker:latest   
```

However I was able to reproduce using #443 (which is mainlining the stuff from the preview). I'll highlight this issue there to be resolved as part of the mainlining effort.

**Comment by silvin-lubecki** [comment]
Hi @tullo thank you for filling this issue, but as @ijc said, this was already fixed on master branch with #441 ! Will close this issue, but if you spot something else feel free to re-open it 👍

**Comment by tullo** [comment]
Hi @silvin-lubecki I see that you addressed the namespace - all lowercase, but isn't the error msg complaining about the slash `/` which is not part of the reg-ex pattern it complains about.
